omnixai/explainers/ranking/agnostic/validity.py

```diff
@@ -0,0 +1,215 @@
+#
+# Copyright (c) 2022 salesforce.com, inc.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+# For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+#
+"""
+Ranking Explainer for tabular data.
+"""
+import numpy as np
+from typing import Callable, List
+from ...base import ExplainerBase
+from ....data.tabular import Tabular
+import scipy
+import itertools
+import pandas as pd
+
+from ....explanations.ranking.agnostic.validity import ValidExplanation
+
+
+class ValidityRankingExplainer(ExplainerBase):
+    """
+    Ranking Explainer for Tabular Data.
+    """
+
+    explanation_type = "local"
+    alias = ["rank"]
+
+    def __init__(
+        self,
+        training_data: Tabular,
+        features: List,
```

Comment:
It is better to change param "features" to "ignored_features" and provide a default value None for it (meaning all the features are considered in the algorithm). The reason is that usually we choose more features to analyze than features to be ignored. So "ignored_features" allows users to set less param values.

Reply:
Then we have self.features = [f for f in training_data.columns if f not in ignored_features].

Reply:
In our usecase, the ignored_features list can be quite large. I think it is better to specify which features require analysis than specifying which ones do not.